File: my-app/src/classes/class.ts

class Cookie {
    private _color: string
    constructor(color: string) {
        this._color = color
    }
    get color(): string {
        return this._color
    }
    set color(newColor: string) {
        this._color = newColor
    }
}

const c1 = new Cookie("blue")
const c2 = new Cookie("black")
console.log(c1.color)
console.log(c2.color)
c2.color = "white"
console.log(c2.color)

File: rustapp/src/bin/classes/class.rs
struct Cookie {
    color: String,
}
impl Cookie {
    fn new(color: &str) -> Self {
        Cookie {
            color: color.to_string(),
        }
    }
    fn get_color(&self) -> &str {
        &self.color
    }
    fn set_color(&mut self, new_color: &str) {
        self.color = new_color.to_string()
    }
}

fn main() {
    println!("hello from rusting");
    let c1 = Cookie::new("black");
    let mut c2 = Cookie::new("white");
    println!("{}", c1.get_color());
    println!("{}", c2.get_color());
    c2.set_color("blue");
    println!("{}", c2.get_color());
}
